x-pack/plugins/apm/server/lib/settings/apm_indices/get_apm_indices.ts

```diff
@@ -64,19 +68,38 @@ export async function getApmIndices({
       savedObjectsClient
     );
     const apmIndicesConfig = getApmIndicesConfig(config);
-    return merge({}, apmIndicesConfig, apmIndicesSavedObject);
+    return mergeApmIndicesConfigs(apmIndicesConfig, apmIndicesSavedObject);
```

Comment:
Why do we need a custom merge function?

Reply:
the custom merge is able to support the deprecated + new APM index configs giving preference to the new config path. A plain merge cannot accomplish this.

Reply:
after doing a migration of the saved objects mapping, i removed the custom merge.